lib/int3.h

```diff
@@ -82,19 +82,7 @@ class int3
 
 	constexpr bool operator<(const int3 & i) const
 	{
-		if (z < i.z)
-			return true;
-		if (z > i.z)
-			return false;
-		if (y < i.y)
-			return true;
-		if (y > i.y)
-			return false;
-		if (x < i.x)
-			return true;
-		if (x > i.x)
-			return false;
-		return false;
+		return z < i.z || y < i.y || x < i.x;
```

Comment:
hm, I'm thinking maybe this is not equivalent to initial implementation which can introduce problems:
`A(0, 0, 1) < B(1, 0, 0)`
Before change: false
After change: true

Reply:
equivalent condition looks following:
`return z < i.z || (z == i.z && (y < i.y || (y == i.y && x < i.x)));`